contracts/contracts/vault/VaultCore.sol

```diff
@@ -112,8 +112,9 @@ contract VaultCore is VaultStorage {
      * @dev Mint OUSD for OUSD Meta Strategy
      * @param _amount Amount of the asset being deposited
      *
-     * Notice: can't use nonReentrant modifier since BaseCurveStrategy's deposit
-     * already has that modifier present
+     * Notice: can't use `nonReentrant` modifier since the `mint` function can
```

Comment:
If we ever do the `mint -> allocate -> mintForStrategy` route we are doomed from allowing users to control funds moving into or out of a manipulable strategy.

I think `reallocate -> mintForStrategy` is the more likely path.

Reply:
Agree and that though did cross my mind. I agree and expect this to never happen on mainnet, because of the reason you point out. Test suite on the other hand is using this exact path. And once we roll out the `strategy testing framework` I expect all strategies tested will need to support `mint -> allocate -> mintForStrategy`.

If we were to only use `reallocate -> mintForStrategy` path we could actually have the non reentrant modifier on `mintForStrategy` since `reallocate` doesn't have non reentrancy guard. 

@DanielVF could you check the comment on `burnForStrategy` as well? That one potentially collides with redeem's non reentrancy. (that again probably won't be part of any mainnet setups).

Reply:
I'm very concerned about having comments in the code mentioning the necessity of supporting a series of action that are a near guarantee of a protocol funds loss. If some future person is reading through the code looking to understand the contracts, they could think this was an important good thing. That's why I would rather these two comments have the slightly longer, but expected in actual use situtation.